tests/scene/test_timer.h

```diff
@@ -145,6 +144,43 @@ TEST_CASE("[SceneTree][Timer] Check Timer process callback") {
 	memdelete(test_timer);
 }
 
+class TestTimerSignals : public Object {
+	GDCLASS(TestTimerSignals, Object);
+
+public:
+	void timer_timeout();
+
+protected:
+	static void _bind_methods();
+};
+
+void TestTimerSignals::timer_timeout() {
+	CHECK_MESSAGE(false, "this should be called!!!");
+}
+
+void TestTimerSignals::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("timer_timeout"), &TestTimerSignals::timer_timeout);
+}
+
+TEST_CASE("[SceneTree][Timer] Check Timer timeout") {
+	Timer *test_timer = memnew(Timer);
+	TestTimerSignals test_signals;
```

Comment:
Not sure how to fix, but maybe you need to memnew this struct as well?

Reply:
I believe it isnt possible to test the signals of nodes in unit tests, i dont see any other node signals being tested.
Ill ask in the dev chat as well, thanks.

Reply:
it is possible to test the signals with the SIGNAL_WATCH and SIGNAL_CHECK macros.